### PR TITLE
Add v1 and v2 badges

### PR DESCRIPTION
**Pull request by ErikSchierboom**
This PR:

- Adds a badge for v1 users
- Adds a badge for v2 users
- Fixes the pluralization of badge modal

The badges can be awarded via this script (please double-check):

```ruby
last_v1_user = User.where('created_at < ?', Time.new(2018, 7, 13, 10, 1, 29, 0)).order('id DESC').first.id
last_v2_user = User.where('created_at < ?', Time.new(2021, 9, 1, 0, 0, 0, 0)).order('id DESC').first.id

User.where.not(confirmed_at: nil).where('id <= ?', last_v1_user).find_each do |user|    
  AwardBadgeJob.perform_later(user, :v1, send_email: false)
end

User.where.not(confirmed_at: nil).where('id <= ?', last_v2_user).find_in_batches do |batch|
  batch.each do |user|
    AwardBadgeJob.perform_later(user, :v2, send_email: false)
  end

  sleep(5)
end
```

### CODE REVIEW

File: app/models/badges/v1_badge.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    def award_to?(user)
+      return false if user.confirmed_at.nil?
+
+      user.created_at < V2_RELEASE_DATE
+    end
+
+    def send_email_on_acquisition?
+      false
+    end
```
Comment (ErikSchierboom) [inline app/models/badges/v1_badge.rb:16]: Does this prevent the email from being sent?

File: app/models/badges/v2_badge.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      return false if user.confirmed_at.nil?
+
+      user.created_at < V3_RELEASE_DATE
+    end
+
+    def send_email_on_acquisition?
+      false
+    end
+
+    V3_RELEASE_DATE = Time.new(2021, 9, 1, 0, 0, 0, 0).freeze
```
Comment (ErikSchierboom) [inline app/models/badges/v2_badge.rb:18]: I've just set the time to the first second, as we had frozen the website before launch anyway so anyone that registered that day must be a v3 user.

File: app/models/badges/v1_badge.rb (hunk 1 above)
Comment (iHiD) [reply app/models/badges/v1_badge.rb:16]: Yes.

File: app/javascript/components/modals/BadgeModal.tsx [hunk 3]
```diff
@@ -36,7 +37,9 @@ export const BadgeModal = ({
       </div>
       <div className="num-awardees text-p-base">
         <GraphicalIcon icon="students" />
-        <strong>{badge.numAwardees}</strong> members have earned this badge.
+        <strong>{badge.numAwardees}</strong>{' '}
+        {pluralize('member', badge.numAwardees)}{' '}
+        {pluralize('has', badge.numAwardees)} earned this badge.
```
Comment (iHiD) [inline app/javascript/components/modals/BadgeModal.tsx:42]: I figure that no badge will only ever go to one person, which is why I didn't do this FYI. But I'm happy with the change :)

**Comment by iHiD** [comment]
## v1.svg

```
<svg width="384" height="384" viewBox="0 0 384 384" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M244.1 101.5C244.1 101.5 247.4 135.3 225.8 153.6C217.1 148.1 206.2 145 193.3 145C181.4 145 169.7 148.4 159.6 154.7C136.5 136.6 140 101.5 140 101.5C114.4 151.4 141.3 170.9 141.9 171.4C134.8 181.6 130.4 194.5 130.4 209.6C130.4 250.1 160.1 274 197.1 274C216 274 232.1 267.7 244.9 259.7L236.1 243.9C225 250.4 213.7 254.5 200.4 254.5C175.7 254.5 157.6 240.2 156.1 216H248.7C249.2 213 250 208.2 250 203.2C250 191.1 247.4 180.3 242.4 171.4C244.8 169.5 268.7 149.5 244.1 101.5V101.5ZM155.8 198.8C159.1 176.9 175.4 164.6 193.8 164.6C214.9 164.6 226.8 176.9 226.8 198.8H155.8ZM192 384C86.1 384 0 297.9 0 192C0 86.1 86.1 0 192 0C297.9 0 384 86.1 384 192C384 297.9 297.9 384 192 384ZM192 24.9C99.8 24.9 24.9 99.9 24.9 192C24.9 284.2 99.9 359.1 192 359.1C284.2 359.1 359.1 284.1 359.1 192C359.1 99.8 284.2 24.9 192 24.9Z" fill="#770428"/>
</svg>
```

## v2.svg

```
<svg width="33" height="33" viewBox="0 0 33 33" fill="none" xmlns="http://www.w3.org/2000/svg">
<g clip-path="url(#clip0_1062_23558)">
<path d="M18.3193 20.4827C18.3193 21.6892 17.2747 22.4432 16.3496 22.4432C15.4244 22.4432 14.3144 22.0683 14.3144 20.4827V19.7115H13.0957V20.4827C13.0957 22.2387 14.556 23.6669 16.3496 23.6669C18.1431 23.6669 19.6034 22.2387 19.6034 20.4827V19.7115H18.3193V20.4827Z" fill="#130B43"/>
<path d="M8.08439 16.7588C8.08439 15.6066 9.08367 14.6696 10.3113 14.6696C11.5383 14.6696 12.5376 15.6074 12.5376 16.7588H13.9402C13.9402 14.8803 12.3132 13.3525 10.3113 13.3525C8.30956 13.3525 6.68164 14.8803 6.68164 16.7588H8.08439Z" fill="#130B43"/>
<path d="M19.6449 16.7588C19.6449 15.6066 20.6441 14.6696 21.8711 14.6696C23.0988 14.6696 24.098 15.6074 24.098 16.7588H25.5008C25.5008 14.8803 23.8728 13.3525 21.8711 13.3525C19.8692 13.3525 18.2422 14.8803 18.2422 16.7588H19.6449Z" fill="#130B43"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M27.7647 12.29V7.76222C27.7647 6.23636 27.5792 5.17768 27.2082 4.58613C26.8372 3.99458 26.0649 3.53849 24.8909 3.21784C24.5365 3.11833 24.3594 2.95801 24.3594 2.73687C24.3594 2.43833 24.758 2.28906 25.5554 2.28906H26.768C28.0858 2.28906 29.0853 2.69816 29.7664 3.51638C30.4475 4.33459 30.788 5.52872 30.788 7.09881V11.8754C30.788 12.9811 30.9071 13.8325 31.1452 14.4295C31.3832 15.0266 31.8235 15.585 32.4657 16.1047C32.6541 16.2594 32.7482 16.3866 32.7482 16.4861C32.7482 16.5856 32.6541 16.7127 32.4657 16.8676C31.8124 17.3983 31.3694 17.9594 31.1368 18.551C30.9043 19.1425 30.788 19.9911 30.788 21.0968V25.8568C30.788 27.4379 30.4447 28.6376 29.758 29.4559C29.0715 30.274 28.0748 30.6831 26.768 30.6831H25.5554C25.0349 30.6831 24.7054 30.6583 24.567 30.6085C24.4286 30.5587 24.3594 30.4344 24.3594 30.2354C24.3594 29.9589 24.5421 29.7765 24.9075 29.6881L25.572 29.5388C27.0338 29.207 27.7647 27.7641 27.7647 25.21V20.6822C27.7647 18.9683 28.18 17.7577 29.0106 17.05L29.3262 16.7847C29.4812 16.652 29.5587 16.5525 29.5587 16.4861C29.5587 16.4087 29.4812 16.3092 29.3262 16.1876L29.0106 15.9222C28.18 15.2145 27.7647 14.0038 27.7647 12.29Z" fill="#130B43"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M4.7315 12.251V7.72316C4.7315 6.1973 4.917 5.13862 5.28798 4.54707C5.65898 3.95552 6.4314 3.49943 7.60529 3.17878C7.95967 3.07927 8.13686 2.91894 8.13686 2.6978C8.13686 2.39927 7.73818 2.25 6.94083 2.25H5.72819C4.41034 2.25 3.41089 2.6591 2.72981 3.47732C2.04874 4.29552 1.7082 5.48966 1.7082 7.05974V11.8363C1.7082 12.942 1.58916 13.7934 1.35106 14.3905C1.11296 14.9876 0.67276 15.5459 0.0304444 16.0656C-0.157817 16.2204 -0.251953 16.3475 -0.251953 16.447C-0.251953 16.5465 -0.157817 16.6737 0.0304444 16.8285C0.683841 17.3593 1.1268 17.9204 1.35937 18.512C1.59193 19.1035 1.7082 19.9521 1.7082 21.0578V25.8178C1.7082 27.3989 2.05151 28.5986 2.73812 29.4167C3.42473 30.235 4.42141 30.6441 5.72819 30.6441H6.94083C7.46133 30.6441 7.79079 30.6192 7.92921 30.5694C8.06765 30.5197 8.13686 30.3953 8.13686 30.1963C8.13686 29.9199 7.95413 29.7374 7.58868 29.6489L6.92422 29.4997C5.4624 29.168 4.7315 27.7251 4.7315 25.171V20.6431C4.7315 18.9293 4.31621 17.7186 3.48564 17.011L3.17002 16.7455C3.01498 16.6129 2.93746 16.5133 2.93746 16.447C2.93746 16.3697 3.01498 16.2702 3.17002 16.1485L3.48564 15.8831C4.31621 15.1755 4.7315 13.9648 4.7315 12.251Z" fill="#130B43"/>
</g>
<defs>
<clipPath id="clip0_1062_23558">
<rect width="32.4969" height="32.9331" fill="white"/>
</clipPath>
</defs>
</svg>
```

File: app/models/badges/v1_badge.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+module Badges
+  class V1Badge < Badge
+    seed "V1",
+      :common,
+      'v1',
+      'Awarded for having joined Exercism v1'
```
Comment (iHiD) [inline app/models/badges/v1_badge.rb]:
```suggestion
      'Awarded for having joined when Exercism was at version 1'
```

(And the same for v2)

File: app/javascript/components/modals/BadgeModal.tsx (hunk 3 above)
Comment (ErikSchierboom) [reply app/javascript/components/modals/BadgeModal.tsx:42]: Well, somebody has to be the first for newer badges that we don't bulk award 🤷

**Comment by ErikSchierboom** [comment]
The icon in the v1 badge looks a little smaller than the other icons:

image

**Comment by ErikSchierboom** [comment]
@iHiD Looking good:

image

**Pull request merged**